tests/functional/adapter/test_custom_schema.py

```diff
@@ -0,0 +1,112 @@
+from abc import ABC, abstractmethod
+
+import pytest
+from dbt.tests.util import run_dbt, run_sql_with_adapter
+
+seed_csv = """
+id,name,date
+1,Easton,1981-05-20 06:46:51
+2,Lillian,1978-09-03 18:10:33
+3,Jeremiah,1982-03-11 03:59:51
+4,Nolan,1976-05-06 20:21:35
+""".lstrip()
+
+
+class CustomSchemaBase(ABC):
+    """
+    This test is meant to ensure that Trino table, view, incremental materialization
+    works as expected for custom schemas
+    """
+
+    @property
+    def schema(self):
+        return "default"
+
+    # set custom schema name
+    custom_schema_name = "very_custom_schema_name"
+
+    @property
+    @abstractmethod
+    def table_type(self):
+        return "default"
+
+    @property
+    @abstractmethod
+    def materialization(self):
+        return "default"
+
+    # define model
+    def custom_schema_model(self, materialization, custom_schema_name):
```

Comment:
`custom_schema_name` seems to be constant, maybe just use `self.custom_schema_name` instead of using a parameter?

Reply:
changed